validate-modules - Use error codes instead of numbers

samdoran opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Add names for each of the error codes used by `validate-modules`.
Change the display message to include `name (code)`

Fixes #59405
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request


##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
`test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py`

samdoran committed:
validate-modules - Use error codes instead of numbers

samdoran commented:
These names are highly up for debate. I attempted to stick with the mindset of describing the check not the error. Feel free to leave suggestions for improving the error code names.

mattclay commented:
I don't see any reason to keep the existing numeric codes. We can just replace them with the names.

mattclay commented on test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py:
```diff
@@ -501,13 +510,15 @@ def _find_module_utils(self, main):
                             if self._is_new_module():
                                 self.reporter.error(
                                     path=self.object_path,
+                                    name='FIXME',
```
I assume this and the other `FIXME` names are placeholders for names not yet decided?

samdoran replied:
Yes.

mattclay commented on test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py:
```diff
@@ -654,6 +671,7 @@ def _ensure_imports_below_docs(self, doc_info, first_callable):
                         if future_import.name not in self.WHITELIST_FUTURE_IMPORTS:
                             self.reporter.error(
                                 path=self.object_path,
+                                name='allowed-future-imports',
```
This test should be removed from the module validator. We have a future boilerplate sanity check which covers this. I'd remove it in a separate PR though.

mattclay commented on test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py:
```diff
@@ -918,6 +948,7 @@ def _validate_docs(self):
                     # for error in errors:
                     #     self.reporter.error(
                     #         path=self.object_path,
+                    #         name='ansible-metadata-format',
```
Was the YAML feature removed from the spec? If so, this whole commented block of code can be removed in a separate PR.

mattclay commented on test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py:
```diff
@@ -362,6 +364,7 @@ def _check_interpreter(self, powershell=False):
         if not self.text.startswith('#!/usr/bin/python'):
             self.reporter.error(
                 path=self.object_path,
+                name='interpreter-line-python',
```
I think we can remove this check in a separate PR, as the shebang sanity test covers this.

mattclay commented on test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py:
```diff
@@ -354,6 +355,7 @@ def _check_interpreter(self, powershell=False):
             if not self.text.startswith('#!powershell\n'):
                 self.reporter.error(
                     path=self.object_path,
+                    name='interpreter-line-powershell',
```
I think we can remove this check in a separate PR, as the shebang sanity test covers this.

samdoran committed:
Add a few missing error names and remove commented out check

samdoran committed:
Update error code format

I'm not sure if this change is correct

samdoran committed:
Correct a few error codes

Fix duplicate/incorrect numbers and names

samdoran committed:
Revert "Update error code format"

This reverts commit 09f08787161164c1111218b33cbbd0e48cbe7292.

samdoran committed:
Add colon after error code

This makes it easier to distinguish the error code from the message, which is more important now that the error code is a string rather than a number.

samdoran committed:
Remove name and replace code with string

This removes all traces of the numeric codes. We will need to update docs to reflect this change.

samdoran committed:
More cleanup from removing the error name parameter

samdoran committed:
Update docs

samdoran committed:
Rework the table it docs

samdoran committed:
Last round of name bikeshedding

# Conflicts:
#	test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

samdoran committed:
A few more name changes

# Conflicts:
#	test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py
#	test/sanity/ignore.txt

samdoran committed:
# This is a combination of 2 commits.
# This is the 1st commit message:

A few more name changes

# Conflicts:
#	test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py
#	test/sanity/ignore.txt

# This is the commit message #2:

Bikeshedding on the error names

samdoran committed:
Add changelog

mattclay commented:
Sanity tests passed. Other failures are unrelated. Merging quickly to avoid more rebase issues.
